pkg/agent/apiserver/handlers/networkpolicy/handler.go

```diff
@@ -48,15 +49,68 @@ func HandleFunc(aq agentquerier.AgentQuerier) http.HandlerFunc {
 		} else {
 			nps = npq.GetNetworkPolicies(npFilter)
 		}
-
-		obj = cpv1beta.NetworkPolicyList{Items: nps}
+		npSorter := &NPSorter{
+			networkPolicies: nps,
+			sortBy:          r.URL.Query().Get("sort-by"),
+		}
+		sort.Sort(sort.Reverse(npSorter))
+		obj = cpv1beta.NetworkPolicyList{Items: npSorter.networkPolicies}
 
 		if err := json.NewEncoder(w).Encode(obj); err != nil {
 			http.Error(w, "Failed to encode response: "+err.Error(), http.StatusInternalServerError)
 		}
 	}
 }
 
+var (
+	sortByEffectivePriority = "effectivePriority"
+	// TODO: this is a hack, but works for now. 251 is a tierPriority value in
+	//  between the application tier and the baseline tier, which can be used
+	//  to sort policies by tier.
+	effectiveTierPriorityK8sNP = int32(251)
+)
+
+type NPSorter struct {
+	networkPolicies []cpv1beta.NetworkPolicy
+	sortBy          string
+}
+
+func (nps *NPSorter) Len() int { return len(nps.networkPolicies) }
+func (nps *NPSorter) Swap(i, j int) {
+	nps.networkPolicies[i], nps.networkPolicies[j] = nps.networkPolicies[j], nps.networkPolicies[i]
+}
+func (nps *NPSorter) Less(i, j int) bool {
+	var ti, tj int32
+	if nps.networkPolicies[i].TierPriority == nil {
+		ti = effectiveTierPriorityK8sNP
+	} else {
+		ti = *nps.networkPolicies[i].TierPriority
+	}
+	if nps.networkPolicies[j].TierPriority == nil {
+		tj = effectiveTierPriorityK8sNP
+	} else {
+		tj = *nps.networkPolicies[j].TierPriority
+	}
+	pi, pj := nps.networkPolicies[i].Priority, nps.networkPolicies[j].Priority
+	switch nps.sortBy {
+	case sortByEffectivePriority:
+		if ti == tj {
+			if (pi == nil && pj == nil) || (pi != nil && pj != nil && *pi == *pj) {
+				return nps.networkPolicies[i].Name > nps.networkPolicies[j].Name
+			}
+			// If two NetworkPolicies are in the same tier and have different priorities
+			// (K8s NetworkPolicy will not apply since pi == pj == nil), the priorities
+			// of those NetworkPolicies must not be nil.
+			return *pi > *pj
+		}
+		return ti > tj
```

Comment:
This leads to less indent:
```
if ti != tj {
    return ti > tj
}
if pi != nil && pj != nil && *pi == *pj {
    return *pi > *pj
}
fallthrough
```

Reply:
and better to include L83-L94 to sortByEffectivePriority case as they are not related to other cases.

Reply:
Thanks for the suggestion.